src/compile.rs

```diff
@@ -210,6 +213,38 @@ impl Compiler {
         Ok(())
     }
 
+    fn compile_conditional<F>(&mut self, mut handle_child: F) -> Result<()>
+    where
+        F: FnMut(&mut Compiler, usize) -> Result<()>,
+    {
+        self.b.add(Insn::BeginAtomic);
```

Comment:
It's kinda explained in the `EndAtomic` below, but it would be nice to have a comment here explaining why we need to use atomic here.